worker/src/lib/util/cache/cacheFunctions.ts

```diff
@@ -132,53 +132,58 @@ export async function getCachedResponse(
   request: HeliconeProxyRequest,
   settings: { bucketSize: number },
   cacheKv: KVNamespace,
-  cacheSeed: string | null
+  cacheSeed: string | null,
 ): Promise<Response | null> {
-  const { requests: requestCaches, freeIndexes } = await getMaxCachedResponses(
-    request,
-    settings,
-    cacheKv,
-    cacheSeed
-  );
-  if (freeIndexes.length > 0) {
-    return null;
-  } else {
-    const cacheIdx = Math.floor(Math.random() * requestCaches.length);
-    const randomCache = requestCaches[cacheIdx];
-    const cachedResponseHeaders = new Headers(randomCache.headers);
-    cachedResponseHeaders.append("Helicone-Cache", "HIT");
-    cachedResponseHeaders.append(
-      "Helicone-Cache-Bucket-Idx",
-      cacheIdx.toString()
-    );
-
-    const cachedStream = new ReadableStream({
-      start(controller) {
-        let index = 0;
-        const encoder = new TextEncoder();
-
-        function pushChunk() {
-          if (index < randomCache.body.length) {
-            const chunk = encoder.encode(randomCache.body[index]);
-            controller.enqueue(chunk);
-            index++;
-            setTimeout(pushChunk, 10);
-          } else {
-            controller.close();
+  const CACHE_TIMEOUT = 2000;
+
+  try {
+    const { requests: requestCaches, freeIndexes } = await Promise.race([
```

Comment:
Inspired by this `.race`, we could also have cache-fetching race against an actual call to the provider. Would increase api costs though.

Reply:
This is a really unique idea, however, it should never be the case that cache checking is > the provider. Like cache should be REALLY faster much much faster than not using the cache. I think Cloudflare KV is just buggy so we need to add some logic here to make sure stuff doesnt break.